src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java

```diff
@@ -1856,6 +1857,81 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
 		}
 	}
 
+	@api
+	public static class hanging_place extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "hanging_place";
+		}
+
+		@Override
+		public String docs() {
+			return "{type: <macro> The entity type of the entity | id: <macro> The entity ID of the entity"
+					+ " | player: <macro> The player triggering the event | world: <macro> The world the entity is in}"
+					+ " This event is called when a player attempts to place an item frame, painting, or leash."
+					+ " {id: The entity ID of the entity. | type: The type of the hanging entity; can be ITEM_FRAME,"
+					+ " PAINTING, or LEASE_HITCH | location: Where the entity was placed. |"
+					+ " player: The name of the place which placed this entity. }"
+					+ " {}"
+					+ " {}";
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public boolean matches(Map<String, Mixed> prefilter, BindableEvent event) throws PrefilterNonMatchException {
+			if(event instanceof MCHangingPlaceEvent) {
+				MCHangingPlaceEvent hangingPlaceEvent = (MCHangingPlaceEvent) event;
+				MCHanging hanging = hangingPlaceEvent.getEntity();
+				Prefilters.match(prefilter, "type", hanging.getType().name(), PrefilterType.MACRO);
+				Prefilters.match(prefilter, "id", hanging.getUniqueId().toString(), PrefilterType.MACRO);
```

Comment:
I'm pretty sure the code will never know the id of the placed entity. So this prefilter is unnecessary.